services/datamanager/builtin/builtin.go

```diff
@@ -504,69 +473,31 @@ func (svc *builtIn) Update(ctx context.Context, cfg *config.Config) error {
 		}
 	}
 
-	toggledCaptureOff := (svc.captureDisabled != svcConfig.CaptureDisabled) && svcConfig.CaptureDisabled
 	svc.captureDisabled = svcConfig.CaptureDisabled
-	var allComponentAttributes []dataCaptureConfig
-
 	// Service is disabled, so close all collectors and clear the map so we can instantiate new ones if we enable this service.
-	if toggledCaptureOff {
+	if svc.captureDisabled {
```

Comment:
[nit] I find it a little confusing that we do captureDisabled logic here, some other stuff, and then !captureDisabled logic further below. I think it would make more sense to have it so 

```
	allComponentAttributes, err := buildDataCaptureConfigs(cfg)
	if err != nil {
		return err
	}
	svc.captureDir = svcConfig.CaptureDir


       svc.captureDisabled = svcConfig.CaptureDisabled
       if svc.captureDisabled {
       ...
       } else
       ...
       }
```

Reply:
good suggestion, changed!